asset-transfer-secured-agreement/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,434 @@
+        // This will fail as the owner is Org1
+        await transferAsset(assetKey, mspIdOrg2,mspIdOrg2, 100,contractOrg2);
+
+        // Org1 will transfer the asset to Org2
+        // This will now complete as the sell price and the bid price are the same
+        await transferAsset(assetKey, mspIdOrg1,mspIdOrg2, 100, contractOrg1);
+
+        // Read the public details by  org1.
+        await readAsset(assetKey, mspIdOrg2, contractOrg1,mspIdOrg1);
+
+        // Read the public details by  org2.
+        await readAsset(assetKey, mspIdOrg2, contractOrg2,mspIdOrg2);
+
+        // Org2 should be able to read the private data details of this asset
+        await readPrivateAssetAfterTransfer(assetKey, mspIdOrg2, contractOrg2);
+
+        // Org1 should not be able to read the private data details of this asset,expected to fail
+        await readPrivateAssetAfterTransfer(assetKey, mspIdOrg1, contractOrg1);
+
+        // This is an update to the public state and requires only the owner to endorse.
+        // Org2 wants to indicate that the items is no longer for sale
+        await changePublicDescriptionAfterTransfer(assetKey,mspIdOrg2,`Asset ${assetKey} owned by ${mspIdOrg2} is NOT for sale`,contractOrg2);
+
+        // Read the public details by  org1.
+        await readAsset(assetKey, mspIdOrg2, contractOrg1,mspIdOrg1);
+
+        // Read the public details by  org2.
+        await readAsset(assetKey, mspIdOrg2, contractOrg2,mspIdOrg2);
+
+    } finally {
+        gatewayOrg1.close();
+        gatewayOrg2.close();
+        clientOrg1.close();
+        clientOrg2.close();
+    }
+}
+
+main().catch(error => {
+    console.error('******** FAILED to run the application:', error);
+    process.exitCode = 1;
+});
+
+async function createAsset(contract:Contract,org:string):Promise<void> {
+    const asset_properties = {
+        object_type: 'asset_properties',
+        asset_id: assetKey,
+        color: 'blue',
+        size: 35,
+        salt: randomBytes
+    };
+    console.log(`${GREEN}--> Submit Transaction: CreateAsset, ${assetKey} as ${org} - endorsed by Org1${RESET}`);
+
+    await contract.submit('CreateAsset', {
+        arguments:[assetKey,`Asset ${assetKey} owned by ${org} is not for sale`],
+        transientData: { asset_properties: JSON.stringify(asset_properties) },
+    });
+
+    console.log(`*** Result: committed, asset ${assetKey} is owned by Org1`);
+}
+
+async function readAsset(assetKey:string, ownerOrg:string, contract:Contract,org:string):Promise<void> {
+    console.log(`${GREEN}--> Evaluate Transactions: ReadAsset as ${org}, - ${assetKey} should be owned by ${ownerOrg}${RESET}`);
+
+    const resultBytes = await contract.evaluateTransaction('ReadAsset', assetKey);
+
+    const resultString = utf8Decoder.decode(resultBytes);
+    if (resultString.length !== 0) {
+
+        const result:{ objectType: string, assetID: string, ownerOrg: string, publicDescription: string } = JSON.parse(resultString);
```

Comment:
It might be more readable to declare an interface for the asset at the module scope rather than include the typing inline, something like:
```typescript
interface Asset {
    objectType: string,
    assetID: string,
    ownerOrg: string,
    publicDescription: string,
}
```

And alongside that, perhaps extract the parsing of an asset to a named function and use that for every method that's parsing an asset.
```typescript
function parseAsset(json: string): Asset {
    return JSON.parse(json);
}
```
You could do some validation within the parseAsset() function that the output contains the expected fields but for the sample I'd be tempted to skip that to keep the code concise.